--- /xen/xen-4.3.2-backup/tools/libxl/libxl_dom.c	2014-02-14 23:22:59.000000000 +0800
+++ /xen/xen-4.3.2/tools/libxl/libxl_dom.c	2014-11-30 14:26:55.596000000 +0800
@@ -263,8 +263,13 @@
     xs_transaction_t t;
     char **ents, **hvm_ents;
     int i, rc;
+	struct xs_permissions rwperm[2];
+    rwperm[0].id=0;
+	rwperm[0].perms=XS_PERM_NONE;
+	rwperm[1].id=(int)domid;
+	rwperm[1].perms=XS_PERM_READ|XS_PERM_WRITE;
 
-    rc = libxl_domain_sched_params_set(CTX, domid, &info->sched_params);
+	rc = libxl_domain_sched_params_set(CTX, domid, &info->sched_params);
     if (rc)
         return rc;
 
@@ -272,7 +277,7 @@
     if (info->cpuid != NULL)
         libxl_cpuid_set(ctx, domid, info->cpuid);
 
-    ents = libxl__calloc(gc, 12 + (info->max_vcpus * 2) + 2, sizeof(char *));
+    ents = libxl__calloc(gc, 12 + (info->max_vcpus * 2) + 4, sizeof(char *));
     ents[0] = "memory/static-max";
     ents[1] = GCSPRINTF("%"PRId64, info->max_memkb);
     ents[2] = "memory/target";
@@ -290,7 +295,8 @@
         ents[12+(i*2)+1] = libxl_bitmap_test(&info->avail_vcpus, i)
                             ? "online" : "offline";
     }
-
+	ents[12+(i*2)]="memory/usage";
+    ents[12+(i*2)+1] = GCSPRINTF("%d", 123);
     hvm_ents = NULL;
     if (info->type == LIBXL_DOMAIN_TYPE_HVM) {
         hvm_ents = libxl__calloc(gc, 3, sizeof(char *));
@@ -308,6 +314,7 @@
     t = xs_transaction_start(ctx->xsh);
 
     libxl__xs_writev(gc, t, dom_path, ents);
+	xs_set_permissions(ctx->xsh,t,libxl__sprintf(gc,"%s/memory/usage",dom_path),rwperm,2);
     if (info->type == LIBXL_DOMAIN_TYPE_HVM)
         libxl__xs_writev(gc, t, dom_path, hvm_ents);
 
